unlock-app/src/components/consumer/Lock.js

```diff
@@ -74,7 +74,7 @@ const mapStateToProps = state => {
     currentKey: state.network.key, // key is a reserved props name
     account: state.network.account,
     lock: state.network.lock,
-    transaction: state.network.account && state.network.account.transaction,
+    transaction: state.network.account && state.network.account.transactions.latest,
```

Comment:
I am worried that you'd need to check for `state.network.account.transactions` too?